### DISCUSSION TITLE
Reason why using WebPack when creating the project

### DISCUSSION THREAD

**Discussion by cfernandezcsp** [Power-Maverick/DataverseDevTools-VSCode#22] [Q&A] [answered]
Can I ask a stupid question?

Why is the tool including WebPack when creating the project?

Wouldn't it be more flexible if only Typescript is included and leave the decision over the other tools to the user?

**Comment by Power-Maverick** [comment, marked as answer]
Using webpacks gives you more control over the output; hence my intention was to create a standard template for the Dataverse TypeScript project. Any reason you would want to use only TypeScript without webpack?

**Reply by filcole** [reply]
I've found [rollup ](https://rollupjs.org/guide/en/) to be marginally faster than webpack and to output non-minimised JavaScript that is more readable/debug friendly.   However, I have found the minimal JavaScript fractionally smaller, and it's what's used in PCFs.  Hard to choose.

[Parcel](https://parceljs.org/) is another possible bundler - but I haven't investigated that one.

**Reply by Power-Maverick** [reply]
Ok so maybe option to select with different settings when creating a TypeScript Dataverse project might be beneficial is what I am hearing?

**Reply by cfernandezcsp** [reply]
In my case, for simplicity, I didn't want to introduce webpack in the project. That's why I went and manually deleted all the webpack config files which is completely fine. No complaints there.

I think, as @filcole mentioned, there could other packing tools out there that are preferred by the users over webpack. If the tool lets you decide if you want to include webpack, or simply stick with the bare minimum which is typescript, then maybe it will fit a wider variety of project scenarios.

**Comment by Power-Maverick** [comment]
@cfernandezcsp & @filcole - what do you think about a feature where the tool allows you to select from an option if you would like to create a project with some bundling tool or just plan TS. And I can keep adding support for the new bundling tool in future releases.
Also, add a setting where dev can set the default Bundling they would like. If that is set, it would bypass the question on project initiation.

**Reply by cfernandezcsp** [reply]
An option to let the user decide to include webpack or not sounds good.
But letting the user decide from a variety of bundling tools is probably too much.
I would focus the development effort in Dataverse specific functionality, but that's just me :)

**Reply by Power-Maverick** [reply]
I hear you and probably you're right. For now I will add the ability to choose from WebPack or Plain TS project just cause I already have it built out.
